sky/optimizer.py

```diff
@@ -390,16 +390,27 @@ def get_reservations_available_resources(
                 node_resources_reprs = ', '.join(f'{node.num_nodes}x ' +
                                                  r.repr_with_region_zone
                                                  for r in node.resources)
+                hints_concat = '\n'.join([
+                    f'{bold}Resource: {repr(resource)}{reset}\n' +
+                    '\n'.join(hint_list)
+                    for resource, hint_list in resource_hints.items()
+                    if hint_list
+                ])
+                hints_formatted = '\n'.join(
+                    map(lambda r: f'        {r}', hints_concat.split('\n')))
```

Comment:
I'm not a fan for the hardcoded spaces here, we should try to make it more obvious why the number of spaces are the way there are. What do you think about something like
```suggestion
                indent_prefix = ' '*len('Hint: ')
                hints_formatted = '\n'.join(
                    map(lambda r: f'{indent_prefix}{r}', hints_concat.split('\n')))
```